internal/resolution/entitysources/catalogdsource.go

```diff
@@ -171,19 +174,36 @@ type declcfgSchema interface {
 
 // TODO: Cleanup once https://github.com/golang/go/issues/45380 implemented
 // We should be able to get rid of the schema arg and switch based on the type passed to this generic
-func fetchCatalogMetadataByScheme[T declcfgSchema](ctx context.Context, cl client.Client, schema, catalogName string) ([]T, error) {
-	cmList := catalogd.CatalogMetadataList{}
-	if err := cl.List(ctx, &cmList, client.MatchingLabels{"schema": schema, "catalog": catalogName}); err != nil {
-		return nil, err
+func fetchCatalogMetadataByScheme[T declcfgSchema](ctx context.Context, url, schema, _ string) ([]T, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
```

Comment:
I might totally be missing something, but why isn't this code using the caching client?

Reply:
This is in the entitysources/ package which I believe was removed as part of #413. I was trying to not do a bunch of refactoring of this logic because it was either going to make it in before this PR or shortly after.